### ISSUE TITLE
False positive for no-restricted-globals rule

### ISSUE THREAD

**Issue by OliverJAsh** [eslint/typescript-eslint-parser#350]
**What version of TypeScript are you using?**
2.4.1

**What version of `typescript-eslint-parser` are you using?**
4.0.0

**What code were you trying to parse?**

``` ts
type foo = {
  location: any;
};
```

and in my ESLint config:

``` js
{
  rules: {
    'no-restricted-globals': [2, 'location'],
  },
}
```

**What did you expect to happen?**
No error

**What happened?**
Unexpected error `[eslint] Unexpected use of 'location'. (no-restricted-globals)`.

---

My understanding of this ESLint rule is it should only complain if the global identifier is used as an expression. http://eslint.org/docs/rules/no-restricted-globals

**Comment by JamesHenry** [comment]
This must be down to the way we currently process type aliases as if they were special vars (in order to get semi rule to, for example). We should definitely fix this.

**Comment by JamesHenry** [comment]
Closing in favour of a single issue here: https://github.com/eslint/typescript-eslint-parser/issues/414

**Comment by j-f1** [comment]
This would probably be fixed with a custom scope class — `no-restricted-globals` uses the scope to find variables to report.
